packages/taquito-http-utils/src/taquito-http-utils.ts

```diff
@@ -83,77 +79,56 @@ export class HttpBackend {
     }
   }
 
-  protected createXHR(): XMLHttpRequest {
-    return new XMLHttpRequestCTOR();
-  }
-
   /**
    *
    * @param options contains options to be passed for the HTTP request (url, method and timeout)
    */
-  createRequest<T>(
+  async createRequest<T>(
     {
       url,
       method,
-      timeout,
+      timeout = defaultTimeout,
       query,
       headers = {},
-      json = true,
-      mimeType = undefined,
+      json = true
     }: HttpRequestOptions,
     data?: object | string
   ) {
-    return new Promise<T>((resolve, reject) => {
-      const request = this.createXHR();
-      request.open(method || 'GET', `${url}${this.serialize(query)}`);
-      if (!headers['Content-Type']) {
-        request.setRequestHeader('Content-Type', 'application/json');
-      }
-      if (mimeType) {
-        request.overrideMimeType(`${mimeType}`);
-      }
-      for (const k in headers) {
-        request.setRequestHeader(k, headers[k]);
-      }
-      request.timeout = timeout || defaultTimeout;
-      request.onload = function () {
-        if (this.status >= 200 && this.status < 300) {
-          if (json) {
-            try {
-              resolve(JSON.parse(request.response));
-            } catch (ex) {
-              reject(new Error(`Unable to parse response: ${request.response}`));
-            }
-          } else {
-            resolve(request.response);
-          }
-        } else {
-          reject(
-            new HttpResponseError(
-              `Http error response: (${this.status}) ${request.response}`,
-              this.status as STATUS_CODE,
-              request.statusText,
-              request.response,
-              url
-            )
-          );
-        }
-      };
+    let resType: 'text' | 'json';
+    let transformResponse = undefined;
 
-      request.ontimeout = function () {
-        reject(new Error(`Request timed out after: ${request.timeout}ms`));
-      };
+    if (!headers['Content-Type']) {
+      headers['Content-Type'] = 'application/json'
+    }
 
-      request.onerror = function (err) {
-        reject(new HttpRequestFailed(url, err));
-      };
+    if (!json) {
+      resType = 'text'
+      transformResponse = [(v: any) => v]
+    } else {
+      resType = 'json';
+    }
+  
+    let response;
+    try {
+      response = await axios.request<T>({
+        url: url + this.serialize(query),
+        method: method ?? 'GET',
+        headers: headers,
+        responseType: resType,
+        transformResponse,
+        timeout: timeout,
+        data: data,
+      });
+    } catch (err: any) {
+      throw new HttpResponseError(
+        `Http error response: (${err.response.status}) ${JSON.stringify(err.response.data)}`,
```

Comment:
On error, it seems that `err.response.data` is sometimes a string, sometimes an object. I see that a change was needed in the following test file: [integration-tests/contract-call-with-xtz-amount.spec.ts](https://github.com/ecadlabs/taquito/pull/1396/files#diff-89b7d3fb841cede12ae4c302aad898f559ab136b37ff7f3ff3467e1c2c863f40). However, `err.response.data` is a string, if not doing JSON.stringify(err.response.data), we keep the same error format as before.

The solution I suggest is to do a validation on the type, and only call `JSON.stringify` if typeof `err.response.data === 'object'`